src/emm/models/repository.py

```diff
@@ -0,0 +1,58 @@
+"""Models for working with git repositories."""
```

Comment:
This contains some models for interfaces for working with repositories in modules. This allows use to share code for the base repo and other module repos, instead of needing separate code paths for each of them.